### PR TITLE
Fix PBMAC1 MAC verification in FIPS mode

### PR DESCRIPTION
**Pull request by beldmit**
The check for fetchability PKCS12KDF doesn't make sense when we have a different MAC mechanism

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
@tomato42 JFYI

**Review by paulidale — APPROVED** [review]

File: apps/pkcs12.c [hunk 1]
```diff
@@ -820,16 +820,6 @@ int pkcs12_main(int argc, char **argv)
...
-        EVP_KDF *pkcs12kdf;
-
-        pkcs12kdf = EVP_KDF_fetch(app_get0_libctx(), "PKCS12KDF",
-                                  app_get0_propq());
-        if (pkcs12kdf == NULL) {
-            BIO_printf(bio_err, "Error verifying PKCS12 MAC; no PKCS12KDF support.\n");
-            BIO_printf(bio_err, "Use -nomacver if MAC verification is not required.\n");
-            goto end;
-        }
-        EVP_KDF_free(pkcs12kdf);
```
Comment (mattcaswell) [inline apps/pkcs12.c:832]: It seems like we are going to lost good error message output in the vast majority of cases where PKCS12KDF is still in use. I assume what we get instead is something much more cryptic? Is there anyway to check up front what KDF is going to be used?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: apps/pkcs12.c (hunk 1 above)
Comment (beldmit) [reply apps/pkcs12.c:832]: Agreed. I've restored the check in case we don't use PBMAC1

File: test/recipes/80-test_pkcs12.t [hunk 2]
```diff
@@ -211,6 +213,28 @@ for my $file ("pbmac1_256_256.good.p12", "pbmac1_512_256.good.p12", "pbmac1_512_
...
+# Test pbmac1 pkcs12 good files, RFC 9579
+    for my $file ("pbmac1_256_256.good.p12", "pbmac1_512_256.good.p12", "pbmac1_512_512.good.p12")
+    {
+        my $path = srctop_file("test", "recipes", "80-test_pkcs12_data", $file);
+        ok(run(app(["openssl", "pkcs12", @prov, "-in", $path, "-password", "pass:1234", "-noenc"])),
+          "test pbmac1 pkcs12 file $file");
+
+        ok(run(app(["openssl", "pkcs12", @prov, "-in", $path, "-info", "-noout",
+		"-passin", "pass:1234"], stderr => "${file}_info.txt")),
+        "test_export_pkcs12_${file}_info");
```
Comment (mattcaswell) [inline test/recipes/80-test_pkcs12.t:234]: Nit: the indentation is slightly off here and inconsistent with the previous similar statement

File: test/recipes/80-test_pkcs12.t [hunk 3]
```diff
@@ -211,6 +213,28 @@ for my $file ("pbmac1_256_256.good.p12", "pbmac1_512_256.good.p12", "pbmac1_512_
...
+    my @prov = ("-provider-path", $provpath,
+                "-provider", $provname);
+    local $ENV{OPENSSL_CONF} = $provconf;
+
+# Test pbmac1 pkcs12 good files, RFC 9579
+    for my $file ("pbmac1_256_256.good.p12", "pbmac1_512_256.good.p12", "pbmac1_512_512.good.p12")
+    {
+        my $path = srctop_file("test", "recipes", "80-test_pkcs12_data", $file);
+        ok(run(app(["openssl", "pkcs12", @prov, "-in", $path, "-password", "pass:1234", "-noenc"])),
+          "test pbmac1 pkcs12 file $file");
```
Comment (mattcaswell) [inline test/recipes/80-test_pkcs12.t:230]: Nit: Should the " really line with the "r" of "run"?

**Review by mattcaswell — APPROVED** [review]
Approved subject to the minor nits being fixed

**Comment by mattcaswell** [comment]
@paulidale - please reconfirm

File: test/recipes/80-test_pkcs12.t (hunk 2 above)
Comment (beldmit) [reply test/recipes/80-test_pkcs12.t:234]: Thanks, fixed
Also fixed tabulation in a different block

File: test/recipes/80-test_pkcs12.t (hunk 3 above)
Comment (beldmit) [reply test/recipes/80-test_pkcs12.t:230]: Aligned accordingly to 
https://github.com/openssl/openssl/blob/master/test/recipes/80-test_pkcs12.t#L261-L264

**Comment by beldmit** [comment]
@mattcaswell is it OK to you from the nits perspective?

**Review by mattcaswell — APPROVED** [review]

**Comment by beldmit** [comment]
The failure seems to be a timeout

**Review by paulidale — APPROVED** [review]

**Review by shahsb — COMMENTED** [review]
Provider compatibility for PRs / fips-releases (pull_request) Skipped

This test has been skipped for 3 times. What's the reason?

**Review by shahsb — APPROVED** [review]

**Comment by beldmit** [comment]
Merged. Thanks for the review.

**Pull request closed without merging**